src/uu/dd/src/parseargs.rs

```diff
@@ -0,0 +1,761 @@
+            }
+            ConvFlag::UCase => {
+                if case.is_some() {
+                    return Err(ParseError::MultipleUCaseLCase);
+                } else {
+                    case = Some(flag)
+                }
+            }
+            ConvFlag::LCase => {
+                if case.is_some() {
+                    return Err(ParseError::MultipleUCaseLCase);
+                } else {
+                    case = Some(flag)
+                }
+            }
+            ConvFlag::Block => match (cbs, unblock) {
+                (Some(cbs), None) => block = Some(cbs),
+                (None, _) => return Err(ParseError::BlockUnblockWithoutCBS),
+                (_, Some(_)) => return Err(ParseError::MultipleBlockUnblock),
+            },
+            ConvFlag::Unblock => match (cbs, block) {
+                (Some(cbs), None) => unblock = Some(cbs),
+                (None, _) => return Err(ParseError::BlockUnblockWithoutCBS),
+                (_, Some(_)) => return Err(ParseError::MultipleBlockUnblock),
+            },
+            ConvFlag::Swab => swab = true,
+            ConvFlag::Sync => sync = true,
+            ConvFlag::NoError => noerror = true,
+            _ => {}
+        }
+    }
+
+    let ctable = parse_ctable(fmt, case);
+    let sync = if sync && (block.is_some() || unblock.is_some()) {
+        Some(b' ')
+    } else if sync {
+        Some(0u8)
+    } else {
+        None
+    };
+
+    Ok(IConvFlags {
+        ctable,
+        block,
+        unblock,
+        swab,
+        sync,
+        noerror,
+    })
+}
+
+/// Parse Conversion Options (Output Variety)
+/// Construct and validate a OConvFlags
+pub fn parse_conv_flag_output(matches: &Matches) -> Result<OConvFlags, ParseError> {
+    let flags = parse_flag_list("conv", matches)?;
+
+    let mut sparse = false;
+    let mut excl = false;
+    let mut nocreat = false;
+    let mut notrunc = false;
+    let mut fdatasync = false;
+    let mut fsync = false;
+
+    for flag in flags {
+        match flag {
+            ConvFlag::Sparse => sparse = true,
+            ConvFlag::Excl => {
+                if !nocreat {
+                    excl = true;
+                } else {
+                    return Err(ParseError::MultipleExclNoCreat);
+                }
+            }
+            ConvFlag::NoCreat => {
+                if !excl {
+                    nocreat = true;
+                } else {
+                    return Err(ParseError::MultipleExclNoCreat);
+                }
+            }
+            ConvFlag::NoTrunc => notrunc = true,
+            ConvFlag::FDataSync => fdatasync = true,
+            ConvFlag::FSync => fsync = true,
+            _ => {}
+        }
+    }
+
+    Ok(OConvFlags {
+        sparse,
+        excl,
+        nocreat,
+        notrunc,
+        fdatasync,
+        fsync,
+    })
+}
+
+/// Parse IFlags struct from CL-input
+pub fn parse_iflags(matches: &Matches) -> Result<IFlags, ParseError> {
+    let mut cio = false;
```

Comment:
I think you can create a `IFlags` struct at the beginning of the function and modify its fields, instead of using these temprorary variables. As I mentioned previously you can derive default for `IFlags` to enable a `IFlags::default()` constructor.

Reply:
I'm hesitant to apply this one. My feelings is that the Default trait is great for tests and other situations when the value is relatively unimportant. However, the parser, however, depends on the flags being set to false when they're not explicitly provided, so

Reply:
Default will not change the default value for bool. This would be a breaking change for rust, so I'm really sure it will never happen. Note that you can also manually implement default (`impl Default for IFlags `) if you need different defaults. However, since all defaults are `false`, deriving default would IMHO be the best solution.

Feel free to keep it this way, the current version is definitely good too.

Reply:
The implementation with Default looks much cleaner. You bring up a good point that I could simply implement Default if I wanted to continue my usual controlled lifestyle. However, I'm going to live on the edge and `#[derive(Default)`. Thanks for the suggestion.